app/src/main/java/com/example/homework1calculator/Calculator.kt

```diff
@@ -0,0 +1,174 @@
+package com.example.homework1calculator
+
+
+class Calculator {
+
+    private val regFindNum = Regex("[-*/+]")
+    private val regFindSigns = Regex("[^-+*/=]")  //(?<=[^-+*/=,{ ])[-+*/=]
+    private val regexSpace = Regex("[\\s]")
+    private val priorityMap = mapOf(
+
+        "+" to 1,
+        "-" to 1,
+        "*" to 2,
+        "/" to 2
+    )
+
+    fun calculate(text: String?): String {
+
+        return text?.let {          // функция расширения .let http://developer.alexanderklimov.ru/android/kotlin/with-apply-also.php
+            if (it.contains('(')) { //
+                calcWithBraces(it)
+            } else {
+                calculateWithNoBraces(it)
+            }
+        } ?: ""
+    }
+
+    //counting without braces
+    private fun calculateWithNoBraces(text: String): String {
+
+        val trimText = text.toTrimText()
+        for (i in trimText) {
+            if (!i.isCorrectSymbol && !i.isDigit()) {
+                throw Exception("You have entered wrong symbol [$i], try again please")
+            }
+        }
```

Comment:
есть функция расширение для строки/массива, также есть функция check, она принимает boolean и если он будет false выкинет исключение
```suggestion
        val incorrectSymbol = trimText.find { !i.isCorrectSymbol && !i.isDigit() }
        if (incorrectSymbol != null) {
            throw Exception("You have entered wrong symbol [$incorrectSymbol], try again please")
        }
```